ui/src/publication/publication-model.ts

```diff
@@ -83,20 +85,53 @@ export type ValidatedPublishCandidates = {
 export type PublicationListingItem = {
     id: PublicationId;
     publishTime: TimeStamp;
-    ratkoPushTime?: TimeStamp;
+    ratkoPushTime: TimeStamp | null;
     status: RatkoPushStatus | null;
     trackNumberIds: LayoutTrackNumberId[];
     hasRatkoPushError: boolean;
 };
 
 export type PublicationDetails = {
     id: PublicationId;
-    publishTime: TimeStamp;
-    trackNumbers: TrackNumberPublishCandidate[];
-    referenceLines: ReferenceLinePublishCandidate[];
-    locationTracks: LocationTrackPublishCandidate[];
-    switches: SwitchPublishCandidate[];
-    kmPosts: KmPostPublishCandidate[];
-    status: RatkoPushStatus | null;
-    ratkoPushTime?: TimeStamp | null;
+    publicationTime: TimeStamp;
+    publicationUser: string;
+    trackNumbers: PublishedTrackNumber[];
+    referenceLines: PublishedReferenceLine[];
+    locationTracks: PublishedLocationTrack[];
+    switches: PublishedSwitch[];
+    kmPosts: PublishedKmPost[];
+    ratkoPushStatus: RatkoPushStatus | null;
+    ratkoPushTime: TimeStamp | null;
+};
+
+export type PublishedTrackNumber = {
+    version: RowVersion;
```

Comment:
Käytetäänkös näitä jossain? Tuo frontin RowVersion taisi olla olio-muotoinen mutta backari taitaa nykyään palauttaa rowversion stringiksi yhdistettynä converterilla... joten epäilisin että ei toimi jos täällä yritetään käyttää niitä id/version kenttiä.

Reply:
Tällä hetkellä ei käytetä mihinkään, tuotu lähinnä sen takia että frontin ja bäkkärin modelit täsmäävät.

Pitäisikö meidän muuttaa RowVersion-oliota nyt niin, että ovat sama sekä frontissa että bäkkärissä?